src/meta/src/stream/stream_manager.rs

```diff
@@ -456,7 +456,7 @@ impl GlobalStreamManager {
             }
 
             let dummy_table_id = table_fragments.table_id();
-
+            // TODO: does this need change? for replace_table
```

Comment:
Yes, the new table has different actor ids, It did not allocate according to the previous one-to-one relationship, but directly called for reallocation. Previously the new replaced table could have a different parallelism from old one, which has been fixed recently. It's acceptable to keep this logic unchanged.

Reply:
Previously I don't know what is "replace table". I found it's for `ALTER TABLE` and sink into table. So it seems no Backfill executor (and also their upstream Source executor) will be involved. 🤔

Reply:
Wait, I'm thinking that even for altering table with connector, it seems the Source actors do not need change? Why do we need to allocate splits……

Reply:
All actors in the table job will be rebuilt including the source executors. I guess we have to retrack the split allocation in the source manager as they're assigned a different set of actor ids.

Reply:
Sure. So I think
- `CREATE TABLE` with connector need retrack
- `CREATE SOURCE` (shared) & MV on sources do not need change.

So no changes need to be made in this PR. Do I understand it correctly?

---

> All actors in the table job will be rebuilt including the source executors. 

I don't fully understand the reason behind this though. Is it feasible not to rebuild the source actor? 🤪 Is it because this makes the implementation easier?

Reply:
> 🤪 Is it because this makes the implementation easier?
I'm afraid so. 🥵